grafana-plugin/src/containers/ServiceNowConfigDrawer/ServiceNowStatusSection.tsx

```diff
@@ -0,0 +1,205 @@
+import React, { useEffect, useReducer } from 'react';
+
+import { SelectableValue } from '@grafana/data';
+import { HorizontalGroup, Select, SelectBaseProps, VerticalGroup } from '@grafana/ui';
+import { observer } from 'mobx-react';
+import { Controller, useFormContext } from 'react-hook-form';
+
+import { Text } from 'components/Text/Text';
+import { ApiSchemas } from 'network/oncall-api/api.types';
+import { useCurrentIntegration } from 'pages/integration/OutgoingTab/OutgoingTab.hooks';
+import { useStore } from 'state/useStore';
+import { OnCallAGStatus } from 'utils/consts';
+
+export interface ServiceNowStatusMapping {
+  [OnCallAGStatus.Firing]?: string;
+  [OnCallAGStatus.Resolved]?: string;
+  [OnCallAGStatus.Silenced]?: string;
+  [OnCallAGStatus.Acknowledged]?: string;
+}
+
+export interface ServiceNowFormFields {
+  additional_settings: ApiSchemas['AlertReceiveChannel']['additional_settings'];
+}
+
+export const ServiceNowStatusSection: React.FC = observer(() => {
+  const { control, setValue, getValues } = useFormContext<ServiceNowFormFields>();
+
+  const [, forceUpdate] = useReducer((x) => x + 1, 0);
+
+  const { alertReceiveChannelStore } = useStore();
+  const currentIntegration = useCurrentIntegration();
+  const { id } = currentIntegration;
+
+  useEffect(() => {
+    (async () => {
+      await alertReceiveChannelStore.fetchServiceNowStatusList({ id });
+      forceUpdate();
```

Comment:
Hmm it's weird we need this since `serviceNowStatusList` is observable and is updated within `fetchServiceNowStatusList`, maybe we can sync today so we quickly take a look together if we can fix it a different way?

Reply:
The `forceUpdate` is to make sure we call the function that returns the list of possible options for each dropdown. Maybe it's not needed here, I can't totally remember why I added here, but it would be needed in other parts because otherwise that logic isn't being triggered.

Reply:
it probably has sth to do with react-hook-form Controller